### PR TITLE
[5.3] Add the ability to disable caching of compiled Blade templates

### PR DESCRIPTION
**Pull request by michaeldyrynda**
Under some circumstances, such as when in development, you may opt to disable caching of compiled Blade templates. This change allows caching to be overwritten via a view configuration option.

### CODE REVIEW

**Comment by taylorotwell** [comment]
I think somewhat proposed this before and I thought it always made more sense to support just setting the "compiled" option in views.php to `null` rather than inventing a new configuration option.

**Comment by michaeldyrynda** [comment]
And just make an `is_null` check against `sourcePath` inside the `isExpired` method? Would need a change in the `compile` method as well I suspect. 

You'd still need to compile the template, you just want the cache to always present as expired, right?

**Comment by taylorotwell** [comment]
Yeah I guess so.

File: src/Illuminate/View/ViewServiceProvider.php [hunk 1]
```diff
@@ -73,8 +73,9 @@ public function registerBladeEngine($resolver)
         // instance to pass into the engine so it can compile the views properly.
         $app->singleton('blade.compiler', function ($app) {
             $cache = $app['config']['view.compiled'];
+            $shouldCache = $app['config']['view.should_cache'] ?: true;
```
Comment (acasar) [inline src/Illuminate/View/ViewServiceProvider.php]: I think this doesn't work. If I set ``should_cache`` to false it's just going to be true again.

**Comment by vlakoff** [comment]
Already refs #12586 and #12587, and maybe there was another topic about this.

File: src/Illuminate/View/ViewServiceProvider.php (hunk 1 above)
Comment (GrahamCampbell) [inline src/Illuminate/View/ViewServiceProvider.php]: yeh, this doesn't work

**Comment by michaeldyrynda** [comment]
You're right @acasar. I did a little more digging based on Taylor's suggestion; setting the `compiled` option to `null` has some deeper reaching side effects, given that the view's `render` method still tries to load the cached contents from disk.

I'll do some more work on this this weekend.

**Comment by michaeldyrynda** [comment]
I've done a little more digging; it seems that with a fresh 5.2 install if you set `view.compiled` to `null` as implemented in #12586, you'll wind up with an exception (which the unit tests don't pick up).

Exception when view.compiled is null

Subsequent commits here should address both that bug, as well as changes discussed in this PR.

**Comment by vlakoff** [comment]
Maybe `cachePath` could be looked for "falsy" rather than just "null"?

**Comment by michaeldyrynda** [comment]
Based on the above, I think that having a new config option may still be worthwhile, given the compiler engine requires including the compiled blade via path (and not passing a compiled string around). By the time execution gets to the engine, it has no knowledge of the compiled path being `null` or otherwise.

**Comment by taylorotwell** [comment]
Not sure I'm going to end up adding this. There are already so many ways you could do it, such as a global middleware that checks env and runs view:clear or something.

**Comment by michaeldyrynda** [comment]
Thanks @taylorotwell - I agree, there's a few other approaches to this. Although on further inspection, there shouldn't even theoretically be a (server-side) caching issues anyway, given the way expirations are handled internally.

**Comment by tillkruss** [comment]
@taylorotwell Just setting `view.compiled` to `null` causes in a fatal error. Is there another way? Disabling the Blade cache on read-only system like Heroku is essential.

**Pull request closed without merging**